### PR TITLE
Add compress_assets plugin

### PR DESCRIPTION
**Pull request by Glandos**
- The plugin, based on every conversation in #288 
- Configuration
- Myself as author
- Docs
- Tests
- Dependencies for Travis

### CODE REVIEW

**Comment by Glandos** [comment]
I still don't understand why there is no coverage reported for brotli and zopfli. The code is tested with pytest…

**Comment by saimn** [comment]
Thanks @Glandos ! From #288 I think it just misses the merging of the progress bars, not critical but it would be good to have this. I'm looking into the coverage issue.

File: tests/test_compress_assets_plugin.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+
+@pytest.mark.parametrize("method,compress_suffix,test_import",
+                         [('gzip', 'gz', None),
+                          ('zopfli', 'gz', 'zopfli.gzip'),
+                          ('brotli', 'br', 'brotli')])
+def test_compress(settings, tmpdir, method, compress_suffix, test_import):
+    if test_import:
+        pytest.importorskip(test_import)
+    compress_options = make_gallery(settings, tmpdir, 'gzip')
```
Comment (saimn) [inline tests/test_compress_assets_plugin.py]: So other methods are not tested because 'gzip' is always used here instead of `method` :sweat_smile:

Comment (Glandos) [reply tests/test_compress_assets_plugin.py]: Ahah. You're right. This happens when it's too late to code.

**Comment by Glandos** [comment]
You're right, I thought about the progress bars only after a good night. I won't edit this through GitHub though, I need to test this on my development environment.

**Comment by saimn** [comment]
Thanks!

**Pull request merged**